etna/datasets/tsdataset.py

```diff
@@ -1123,3 +1159,30 @@ def info(self, segments: Optional[Sequence[str]] = None) -> None:
         # print the results
         result_string = "\n".join(lines)
         print(result_string)
+
+    def to_torch_dataset(
+        self, make_samples: Callable[[pd.DataFrame], Union[Iterator[dict], Iterable[dict]]], dropna: bool = True
+    ) -> "Dataset":
+        """Convert the tsdataset to a torch.Dataset.
+
+        Parameters
+        ----------
+        make_samples:
+            function that takes per segment DataFrame and returns iterabale of samples
+
+        dropna:
+            if ``True``, missing rows are dropped
+
+        Returns
+        -------
+        :
+            torch.Dataset with with train or test samples to infer on
+        """
+        df = self.to_pandas(flatten=True)
+        if dropna:
+            df = df.dropna()  # TODO: Fix this
```

Comment:
As in case of regression models where we drop nans by default, otherwise we have nans in predictions or exception is raised.

For the same reason we drop nans. But current logic will drops intermediate time indexes too and batches would not correctly be generated.